BitcoinApp/CoinViewController/CoinNetworkService.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  CoinNetworkService.swift
+//  BitcoinApp
+//
+//  Created by Zhassulan Aimukhambetov on 12/4/19.
+//  Copyright © 2019 Zhassulan Aimukhambetov. All rights reserved.
+//
+
+import Foundation
+
+class CoinNetworkService {
+    
+    static private let urlApi: String = "https://api.coindesk.com/v1/bpi/currentprice/"
+    
+    static func getCoin (currency: Currency, completion: @escaping (Coin) -> Void) {
+        let urlString = urlApi + currency.name + ".json"
+        guard let url = URL(string: urlString) else { return }
+        NetworkService.shared.getData(url: url) { (json) in
```

Comment:
Instead of using `shared` NetworkService rethink dependency injecting logic. For example, you can make a custom initializer for your CoinNetworkService with `NetworkService` as an argument.